src/org/usfirst/frc/team4915/stronghold/commands/AutoCommand1.java

```diff
@@ -27,104 +27,43 @@ public AutoCommand1(Autonomous.Type type, Autonomous.Strat strat, Autonomous.Pos
+        else
+            addSequential(new LauncherGoToNeutralPositionCommand());
 
     	switch (strat) {
-		case DRIVE_SHOOT_VISION: // sets us up to use vision to shoot a high
-									// goal
-		    //if it is low bar launcher will go to travelPosition
-		if (getLauncherBeginPosition(type) == true){
-		    addSequential(new LauncherGoToTravelPositionCommand());
-            addParallel(new AimLauncherCommand());
-            addSequential(new WaitCommand(.75));
+        case NONE:
+            break;
+
+        case DRIVE_ACROSS:
+            System.out.println("Starting Move Straight");
             addSequential(new AutoDriveStraight(getDistance(type), getSpeed(type)));
-            addSequential(new AutoRotateDegrees( getDegrees(position)));
-            if (ModuleManager.VISION_MODULE_ON) {
-                addSequential(new AutoAimControlCommand(true, true));
-                addParallel(new ArcadeDrive());
-            }
-            break;    
-		}
-		//false: For all other types of barriers
-		//launcher will go to NeutralPosition
-		else{
-		    addSequential(new LauncherGoToNeutralPositionCommand());
-            addParallel(new AimLauncherCommand());
-            addSequential(new WaitCommand(.75));
-			addSequential(new AutoDriveStraight(getDistance(type), getSpeed(type)));
-			addSequential(new AutoRotateDegrees( getDegrees(position)));
-			if (ModuleManager.VISION_MODULE_ON) {
-				addSequential(new AutoAimControlCommand(true, true));
-				addParallel(new ArcadeDrive());
-				addParallel(new VisionAimLauncherCommand());
-                addSequential(new AutoLaunchCommand());
-			}
-			break;
-		}
-		
+            break;
+
+		case DRIVE_SHOOT_VISION: // sets us up to use vision to shoot a high goal
+            addSequential(new AutoDriveStraight(getDistance(type), getSpeed(type)));
+            addSequential(new AutoRotateDegrees( getTurnAngle(position)));
+            addSequential(new AutoAimControlCommand(true, true));
+            addSequential(new AutoVisionDriveAndAim());
+            addSequential(new AutoLaunchCommand());
+            break;
+
 		case DRIVE_SHOOT_NO_VISION:
-		    if (getLauncherBeginPosition(type) == true){
-		        addSequential(new LauncherGoToTravelPositionCommand());
-                addParallel(new AimLauncherCommand());
-                addSequential(new WaitCommand(.75));
-		        System.out.println("Starting Move Straight");
-		        addSequential(new AutoDriveStraight(getDistance(type), getSpeed(type)));
-		        addSequential(new AutoDriveStraight(getDistancePastDefense(position), getSpeed(type)));
-		        addSequential(new AutoRotateDegrees(getDegrees(position)));
+		    addSequential(new AutoDriveStraight(getDistance(type), getSpeed(type)));
+		    addSequential(new AutoDriveStraight(getDistancePastDefense(position), getSpeed(type)));
+		    addSequential(new AutoRotateDegrees(getTurnAngle(position)));
 			if (ModuleManager.INTAKELAUNCHER_MODULE_ON) {
```

Comment:
Perhaps the intake module check should be done before the `switch` so it can fail (or fall back to `DRIVE_STRAIGHT`) if you choose a shooty strategy with a disabled launcher